doc: update the cmake testing instruction

zlargon opened the pull request:
Related Issue: https://github.com/libuv/libuv/issues/2251

The manually cmake testing instruction is incorrect. The test programs should always be executed at the root of libuv folder. I also update the cmake instruction for Windows.

```bash
$ mkdir -p out/cmake ; cd out/cmake ; cmake -DBUILD_TESTING=ON ../..
$ make all test                       # on Unix
$ cmake --build . ; ctest -C Debug    # on Windows

# Or manually run tests:
$ ./out/cmake/uv_run_tests    # shared library build
$ ./out/cmake/uv_run_tests_a  # static library build
```

bnoordhuis commented on README.md:
```diff
@@ -173,10 +173,12 @@ To build with [CMake](https://cmake.org/):
-$ ./uv_run_tests    # shared library build
-$ ./uv_run_tests_a  # static library build
+$ make all test                       # on Unix
+$ cmake --build . ; ctest -C Debug    # on Windows
```
cmake + ctest works on both so we might as well simplify this and drop the `make all test` line.

zlargon replied:
Sounds good! I also will add `--output-on-failure` option to `ctest` because `ctest` would not show any test log whether tests pass or failure by default.

zlargon commented:
Hi, @bnoordhuis 

I have updated the markdown and rebased the commit with the latest branch.

BTW, I found that libuv always test failed with `fs_event_error_reporting` on my mac (10.14), even with latest libuv version `1.28.1`. However, it seems that Jenkins only test macOS with 10.10 and 10.11.

I have reported this issue to here https://github.com/libuv/libuv/issues/2071#issuecomment-482168260. Hope someone could take a look.
Thank you.

refack commented:
FTR: this is the current script used for CI testing:
```bash
#verify that the gcc version
gcc --version || true

curl https://cmake.org/files/v3.11/cmake-3.11.2-Linux-x86_64.tar.gz | tar xzf -

mv cmake-3.11.2-Linux-x86_64 cmake-bin/

mkdir -p out/cmake

cd out/cmake

$WORKSPACE/cmake-bin/bin/cmake -DBUILD_TESTING=ON $WORKSPACE

make -J

make test
```

bnoordhuis commented:
Thanks Leon, landed in 87a116c4.
